addons/point_of_sale/static/src/js/Screens/ReceiptScreen/ReceiptScreen.js

```diff
@@ -137,6 +137,13 @@ export class ReceiptScreen extends AbstractReceiptScreen {
             name: partner ? partner.name : this.orderUiState.inputEmail,
         };
         const order_server_id = this.env.pos.validated_orders_name_server_id_map[orderName];
+        if (!order_server_id) {
+            this.showPopup('ErrorPopup', {
+                title: this.env._t('Unsynced order'),
+                body: this.env._t('This order is not yet synced to server. Make sure it is synced then try again.'),
+            });
```

Comment:
This won't work anymore in master due to the refactor of pos. You need
to first add `this.popup = useService("popup")` in the setup of this component and then do something like `this.popup.add(....)` instead of calling `showPopup`. Check how it is done in `PaymentScreen` to have a concrete example